packages/playwright/src/knapsack-playwright.ts

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env node
+
+import { readFileSync, mkdirSync } from 'fs';
+import { execSync, spawnSync } from 'child_process';
+
+import pkg from '@knapsack-pro/playwright/package.json' with { type: 'json' };
+import {
+  KnapsackProCore,
+  KnapsackProLogger,
+  onQueueFailureType,
+  onQueueSuccessType,
+  TestFile,
+} from '@knapsack-pro/core';
+
+if (process.env.KNAPSACK_PRO_TEST_SUITE_TOKEN_PLAYWRIGHT) {
+  process.env.KNAPSACK_PRO_TEST_SUITE_TOKEN =
+    process.env.KNAPSACK_PRO_TEST_SUITE_TOKEN_PLAYWRIGHT;
+}
+
+async function main() {
+  const knapsackProLogger = new KnapsackProLogger();
+  knapsackProLogger.debug(`Running ${pkg.name}@${pkg.version}`);
+
+  mkdirSync('.knapsack-pro', { recursive: true });
+
+  // 0   1                        2
+  // npx @knapsack-pro/playwright ...
+  const cliArguments = process.argv.slice(2).join(' ');
+  // TODO: Remove from cliArguments the positional arguments. (Options like --grep, --grep-invert can stay.)
+  knapsackProLogger.debug(`cliArguments: ${cliArguments}`);
+
+  // --reporter must be last to overwrite a (potentially) preceeding --reporter
+  const command = `npx playwright test --list ${cliArguments} --reporter=@knapsack-pro/playwright/reporters/list`;
+  knapsackProLogger.debug(`Executing: ${command}`);
+
+  execSync(command, { stdio: 'ignore' }); // This may throw an error
+  const tests = JSON.parse(
+    readFileSync('.knapsack-pro/list.json', 'utf8'),
+  ) as string[]; // This may throw an error
+  knapsackProLogger.debug(`Tests to run: ${tests}`);
+
+  const knapsackPro = new KnapsackProCore(pkg.name, pkg.version, () =>
+    tests.map((testFilePath) => ({ path: testFilePath })),
+  );
+
+  const onSuccess: onQueueSuccessType = async (testFiles: TestFile[]) => {
+    const paths = testFiles.map((testFile) => testFile.path).join(' ');
+    const command = `PWTEST_BLOB_DO_NOT_REMOVE=1 npx playwright test ${cliArguments} ${paths}`;
+    knapsackProLogger.debug(`Executing: ${command}`);
+    spawnSync(command, { shell: true, stdio: 'inherit' });
+    return JSON.parse(readFileSync('.knapsack-pro/batch.json', 'utf8')); // This may throw an error
```

Comment:
question

If the spawned process is killed (let's say Spot Instance terminated the spawned playwright but it did not terminate the main Knapsack Pro process yet), is it possible that `.knapsack-pro/batch.json` exists as a leftover from the previous batch? If so, then we would report a wrong batch.

Perhaps we should ensure `.knapsack-pro/batch.json` is removed after we read it for the first time, or before we spawn a shell process with playwright. And then let the `readFileSync` fail due lack of the file.